lucid/launcher.py

```diff
@@ -245,6 +246,12 @@ def grid_to_dock():
 def main():
     args = parse_arguments()
     kwargs = vars(args)
+    # TODO make configurable
+    timeout = 10
+    EpicsSignalBase.set_defaults(
+        timeout=timeout,
+        connection_timeout=timeout,
```

Comment:
I think I'd recommend the auto monitor option as well:

https://github.com/klauer/las-dispersion-scan/blob/176df2006080523d4ae56ae50c04c0ccd8acbdc5/las_dispersion_scan/bin/gui.py#L64-L72

And we have this in so many repos... probably time for a pcdsutils inclusion?

Reply:
If we have it in a shared spot like pcdsutils, we can look at general env vars for tweakability:
`PCDS_GUI_CA_TIMEOUT` (`PCDSUTILS_GUI_CA_TIMEOUT`?) or something similar

Reply:
I'm :+1: on a `pcdsutils` inclusion.

I can put on the `auto_monitor` option but I'm actually not sure the purpose/result of that. Specifically, we'll be hitting ophyd any time any PV updates? That might be a lot of updates. Something to keep in mind is that the PyDM widgets are separately monitoring as well.

Reply:
For background: `auto_monitor` is a pyepics thing that sets up its subscription mechanism automatically: https://pyepics.github.io/pyepics/pv.html#pv-automonitor-label

I had found that it reduced the amount of timeouts I got from interacting with ophyd signals/devices via `.get()` `.put()` and `.set()` in the above project. You'll also see it used in EpicsMotor, avoiding different-but-similar timeouts that Pete was seeing (https://github.com/bluesky/ophyd/blob/0e8acb3df3d17e0ab7aa2cc924831f4a0c580449/ophyd/epics_motor.py#L42-L64)

You're definitely right that it could affect performance. Perhaps it has no place in the general LUCID config at the moment, unless we continue to have timeout issues after this PR. That then leads to Robert's point as to "LUCID wants this `EpicsSignal` configuration but las-dispersion-scan and btms-ui want a different configuration"... Which then leads me to think we shouldn't have this in pcdsutils 😁

Reply:
I'm going to try it and see if it feels any different at all (and check cpu usage etc.)

Reply:
`auto_monitor=True` had the following affects on the TMO lucid:
- Slightly smaller cpu usage spike during startup
- Higher cpu usage in general after startup (4% -> 7% with no typhos open on psbuild-rhel7-03)

I'm going to err on the side of leaving it as-is because we haven't done a lot of typhos/lucid testing with auto_monitor

Reply:
I forgot to say: the app was exactly as responsive as before RE button clicks etc., but keep in mind that a lot of the PyDM stuff sidesteps ophyd

Reply:
Good call on the testing. I agree with your assessment 👍